explorer/interpreter/type_checker.cpp

```diff
@@ -1997,6 +2009,42 @@ auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
           return handle_binary_arithmetic(Builtins::MulWith);
         case Operator::Mod:
           return handle_binary_arithmetic(Builtins::ModWith);
+        case Operator::BitwiseAnd:
+          // `&` between type-of-types performs constraint combination.
+          // TODO: Should this be done via an intrinsic?
```

Comment:
FWIW, I assume this is about trying to also rewrite & on types. My sense is yes it should be rewritten for consistency, but I'm hesitant to do that while we don't have a `constexpr` equivalent. I'm not sure it would work reliably, even if something happened to work right now.